### ISSUE TITLE
Ionic serve + rollup = main.js not updated

### ISSUE THREAD

**Issue by radzik07** [ionic-team/ionic-app-scripts#1189]
#### Description of the problem:
The behavior is ALMOST the same as in first post of https://github.com/ionic-team/ionic-app-scripts/issues/1092 but the reason may be different. 
This started to happen recently, while upgraded to newest ionic/ionic-app-scripts.
After launching ionic serve the main.js is generated properly. 
After adding alert('OK') to some test1.ts file, the main.js is generated properly.
After adding alert('alert2') to test2.ts file, the main.js is generated properly.
After adding some code that throws a javascript error in test2.ts, then in app navigate to the place and throw the error, and then finally correct the error, the main.js is not updated. After the error was thrown, no change in .ts files will ever generate main.js, even if I correct the mistake. It works only after relaunching ionic serve.

The other thing I have configured is rollup-replace. 

This is the log just after launching `ionic serve --dev-remote`
```
[INFO] Starting app-scripts server: --address 0.0.0.0 --port 8100 --livereload-port 35729 - Ctrl+C to cancel
[22:23:41]  watch started ... 
[22:23:41]  build dev started ... 
[22:23:41]  clean started ... 
[22:23:41]  clean finished in 3 ms 
[22:23:41]  copy started ... 
[22:23:41]  transpile started ... 
ENVIRONMENT environment/_environment.dev-remote
[22:23:44]  transpile finished in 3.41 s 
[22:23:44]  preprocess started ... 
[22:23:44]  deeplinks started ... 
[22:23:44]  deeplinks finished in 99 ms 
[22:23:44]  preprocess finished in 100 ms 
[22:23:44]  rollup started ... 
[22:23:44]  copy finished in 3.62 s 
[22:23:52]  rollup: commonjs-proxy:/Applications/AMPPS/www/story2/node_modules/@angular/core/@angular/core.es5.js has 
            issued a warning: 'default' is not exported by 'node_modules/@angular/core/@angular/core.es5.js' 
[22:23:52]  rollup: commonjs-proxy:/Applications/AMPPS/www/story2/node_modules/@angular/forms/@angular/forms.es5.js has 
            issued a warning: 'default' is not exported by 'node_modules/@angular/forms/@angular/forms.es5.js' 
[22:23:52]  rollup: commonjs-proxy:/Applications/AMPPS/www/story2/node_modules/libphonenumber-js/index.es6.js has 
            issued a warning: 'default' is not exported by 'node_modules/libphonenumber-js/index.es6.js' 
[22:23:52]  rollup: /Applications/AMPPS/www/story2/src/pages/authorized/book/book.ts has issued a warning: 'duration' is 
            not exported by 'node_modules/moment/src/moment.js' 
[22:23:59]  rollup finished in 14.84 s 
[22:23:59]  sass started ... 
[22:24:01]  sass finished in 1.92 s 
[22:24:01]  postprocess started ... 
[22:24:01]  postprocess finished in 18 ms 
[22:24:01]  lint started ... 
[22:24:01]  build dev finished in 20.34 s 
[22:24:01]  watch ready in 20.42 s 
[22:24:01]  dev server running: http://localhost:8100/ 
```

I can see the rollup warnings here. Now, let's add some errored code to .ts file and see the log:
```
[22:27:33]  build started ... 
[22:27:33]  transpile update started ... 
[22:27:33]  transpile update finished in 347 ms 
[22:27:33]  deeplinks update started ... 
[22:27:34]  deeplinks update finished in 204 ms 
[22:27:34]  rollup update started ... 
[22:27:38]  rollup: commonjs-proxy:/Applications/AMPPS/www/story2/node_modules/@angular/core/@angular/core.es5.js has 
            issued a warning: 'default' is not exported by 'node_modules/@angular/core/@angular/core.es5.js' 
[22:27:38]  rollup: commonjs-proxy:/Applications/AMPPS/www/story2/node_modules/@angular/forms/@angular/forms.es5.js has 
            issued a warning: 'default' is not exported by 'node_modules/@angular/forms/@angular/forms.es5.js' 
[22:27:38]  rollup: commonjs-proxy:/Applications/AMPPS/www/story2/node_modules/libphonenumber-js/index.es6.js has 
            issued a warning: 'default' is not exported by 'node_modules/libphonenumber-js/index.es6.js' 
[22:27:38]  rollup: /Applications/AMPPS/www/story2/src/pages/authorized/book/book.ts has issued a warning: 'duration' is 
            not exported by 'node_modules/moment/src/moment.js' 
[22:27:39]  build finished in 5.73 s 
```

This is it. Rollup isn't finished, but says the build is. It is like this everytime I try to edit .ts file later, if the error was done in the past. Very irritating. 
Does these warnings may cause that? They're orange color in my console. I wouldn't assume they could be breaking the build. And, even if I restart the ionic serve, I can see them and I can work with the code as long as I won't do any error.

I totally have no idea, it might not even be a bug in ionic-app-scripts, but in something else... what I can say for sure is that I didnt face that kind of problem for the last 6 months I've been working with Ionic.

When I remove the `"ionic_rollup": "./src/config/rollup.config.js"` from package.json, it is working. This is my temporary "solution" for now.

I tested it on Mac and on Windows - both same behavior.
I currently have no idea what to do about it. I tried a lot of versions of node, of ionic, of app-scripts. 

My config in rollup.config.js
```
var nodeResolve = require('rollup-plugin-node-resolve');
var commonjs = require('rollup-plugin-commonjs');
var globals = require('rollup-plugin-node-globals');
var builtins = require('rollup-plugin-node-builtins');
var json = require('rollup-plugin-json');
var replace = require('rollup-plugin-replace');
var argv = require('yargs').argv;

// Read environment from passed parameter.
var env = 'environment/_environment.dev';
env = (argv.dev !== undefined) ? 'environment/_environment.dev' : env;
env = (argv['dev-remote'] !== undefined) ? 'environment/_environment.dev-remote' : env;
env = (argv.prod !== undefined) ? 'environment/_environment.prod' : env;

console.log('ENVIRONMENT', env);
var rollupConfig = {
	/**
	 * entry: The bundle's starting point. This file will
	 * be included, along with the minimum necessary code
	 * from its dependencies
	 */
	entry: process.env.IONIC_APP_ENTRY_POINT,

	/**
	 * sourceMap: If true, a separate sourcemap file will
	 * be created.
	 */
	sourceMap: true,

	/**
	 * format: The format of the generated bundle
	 */
	format: 'iife',

	/**
	 * dest: the output filename for the bundle in the buildDir
	 */
	dest: process.env.IONIC_OUTPUT_JS_FILE_NAME,

	/**
	 * plugins: Array of plugin objects, or a single plugin object.
	 * See https://github.com/rollup/rollup/wiki/Plugins for more info.
	 */
	plugins: [
		builtins(),
		commonjs(),
		nodeResolve({
			module: true,
			jsnext: true,
			main: true,
			browser: true,
			extensions: ['.js']
		}),
		globals(),
		json(),
		replace({
			exclude: 'node_modules/**',
			'environment/_environment.dev': env
		})
	]

};


module.exports = rollupConfig;
```

My package.json

```
{
  "name": "Story",
  "version": "0.0.0",
  "author": "Ionic Framework",
  "homepage": "http://ionicframework.com/",
  "private": true,
  "scripts": {
    "clean": "ionic-app-scripts clean",
    "build": "ionic-app-scripts build",
    "ionic:build": "ionic-app-scripts build",
    "ionic:serve": "ionic-app-scripts serve"
  },
  "dependencies": {
    "@angular/common": "4.1.3",
    "@angular/compiler": "4.1.3",
    "@angular/compiler-cli": "4.1.3",
    "@angular/core": "4.1.3",
    "@angular/forms": "4.1.3",
    "@angular/http": "4.1.3",
    "@angular/platform-browser": "4.1.3",
    "@angular/platform-browser-dynamic": "4.1.3",
    "@ionic-native/core": "3.12.1",
    "@ionic-native/splash-screen": "3.12.1",
    "@ionic-native/status-bar": "3.12.1",
    "@ionic/storage": "2.0.1",
    "ajv": "^5.2.2",
    "babel-minify": "^0.2.0",
    "cordova-android": "^6.2.3",
    "cordova-ios": "^4.4.0",
    "cordova-plugin-console": "^1.0.5",
    "cordova-plugin-device": "1.1.4",
    "cordova-plugin-splashscreen": "~4.0.1",
    "cordova-plugin-statusbar": "2.2.1",
    "cordova-plugin-whitelist": "1.3.1",
    "cordova-plugin-wkwebview-engine": "git+https://github.com/ionic-team/cordova-plugin-wkwebview-engine.git",
    "ionic-angular": "^3.6.0",
    "ionic-cache": "^2.0.4",
    "ionic-plugin-keyboard": "~2.2.1",
    "ionicons": "~3.0.0",
    "moment": "^2.18.1",
    "ng2-validation": "^4.2.0",
    "node-sass": "^4.5.3",
    "object-keys": "^1.0.11",
    "rxjs": "5.4.0",
    "sw-toolbox": "3.6.0",
    "yargs": "^8.0.2",
    "zone.js": "0.8.12"
  },
  "devDependencies": {
    "@angular/tsc-wrapped": "^4.3.5",
    "@ionic/app-scripts": "^2.1.4",
    "@ionic/cli-plugin-ionic-angular": "1.4.1",
    "rollup-plugin-replace": "^1.1.1",
    "typescript": "2.3.4"
  },
  "cordovaPlugins": [
    "cordova-plugin-whitelist",
    "cordova-plugin-statusbar",
    "cordova-plugin-console",
    "cordova-plugin-device",
    "cordova-plugin-splashscreen",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": [
    "ios",
    {
      "platform": "ios",
      "version": "",
      "locator": "ios"
    }
  ],
  "description": "story2: An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-console": {},
      "cordova-plugin-device": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-whitelist": {},
      "ionic-plugin-keyboard": {},
      "cordova-plugin-wkwebview-engine": {}
    },
    "platforms": [
      "android",
      "ios"
    ]
  },
  "config": {
    "ionic_rollup": "./src/config/rollup.config.js"
  }
}
```

my ionic info
```
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.9.2
    ionic (Ionic CLI) : 3.9.2

local packages:

    @ionic/app-scripts : 2.1.4
    Ionic Framework    : ionic-angular 3.6.0

System:

    Node : v6.11.2
    npm  : 3.10.10 
    OS   : macOS Sierra
```

Nodejs v6.11.2
